R/human_infection.R

```diff
@@ -130,17 +128,15 @@ calculate_infections <- function(
   renderer,
   timestep
   ) {
-  source_humans <- variables$state$get_index_of(
```

Comment:
It seems like a mistake to calculate infections for all humans. This would add a lot of computation especially at higher EIRs.

Could we revert this to how it was?

Reply:
I'm not sure @pwinskill? My understanding is that we need to calculate the superinfections that occur in the D and Tr compartments because those superinfections result in an increase in hypnozoite batch number (although not additional blood stage infection). We could change it back for falciparum at least, including another switch earlier in the code, but vivax might need to be longer...

Reply:
ah, the falciparum model doesn't have superinfection from D or Tr. Also, this actually changes the immunity behaviour of the falciparum model. It probably has to go back to its original behaviour.

If these computations become sufficiently different, it may be cleaner to just have `calculate_vivax_infections` and a `calculate_falciparum_infections` and put a switch in `simulate_infection`

Reply:
Would it work/be neater to keep the infections flow the same for both parasites (ie with 'S', 'A', 'U'), then add an additional function to capture hypnozoite batch number changes for 'D' and 'Tr'?

Reply:
I've rearranged the code to reflect this now. Let me know what you think - I can shove the vivax infections into a new function if you think that would be better.